Sources/Action.swift

```diff
@@ -148,6 +147,16 @@ public protocol ActionProtocol {
 	/// `NoError` can be used.
 	associatedtype Error: Swift.Error
 
+	/// Initializes an action that will be conditionally enabled, and creates a
+	/// SignalProducer for each input.
+	///
+	/// - parameters:
+	///   - enabledIf: Boolean property that shows whether the action is
+	///                enabled.
+	///   - execute: A closure that returns the signal producer returned by
+	///              calling `apply(Input)` on the action.
+	init<P: PropertyProtocol>(enabledIf property: P, _ execute: @escaping (Input) -> SignalProducer<Output, Error>) where P.Value == Bool
```

Comment:
I really like the idea of making `ActionState` more powerful by allowing it to carry an optional value, but I can't think of a way to achieve that without leaking implementation details out via `ActionProtocol`.

That is, I'd like to make a private initialiser, but because `ActionProtocol.init(input:_:)` is in a protocol extension it has to call a public initialiser. Starting to wonder whether the only way to make this safe & atomic is to reintroduce `PropertyAction`, so it can redefine `isEnabled` in terms of `input` while still providing atomic access to the current value.